pkg/occlient/occlient.go

```diff
@@ -1629,21 +1631,42 @@ func (c *Client) GetServerVersion() (*serverInfo, error) {
 	return &info, nil
 }
 
-func (c *Client) ExecCMDInContainer(podName string, cmd []string) (string, error) {
+// ExecCMDInContainer execute command in first container of a pod
+func (c *Client) ExecCMDInContainer(podName string, cmd []string, stdout, stderr io.Writer, stdin io.Reader, tty bool) error {
 
-	// TODO: do this without using 'oc' binary
-	args := []string{
-		"exec",
-		podName,
-		"--",
-		strings.Join(cmd, " "),
+	req := c.kubeClient.CoreV1().RESTClient().
+		Post().
+		Namespace(c.namespace).
+		Resource("pods").
+		Name(podName).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Command: cmd,
+			Stdin:   stdin != nil,
+			Stdout:  stdout != nil,
+			Stderr:  stderr != nil,
+			TTY:     tty,
+		}, scheme.ParameterCodec)
+
+	config, err := c.kubeConfig.ClientConfig()
+	if err != nil {
+		return errors.Wrapf(err, "unable to get Kubernetes client config")
 	}
 
-	output, err := c.runOcComamnd(&OcCommand{args: args})
+	exec, err := remotecommand.NewSPDYExecutor(config, "POST", req.URL())
```

Comment:
For example, this function name is bit confusing

Reply:
This function is from client-go library. But I've tried to add comment with explanation.